src/EhsnPlugin/Parser.cs

```diff
@@ -1,25 +1,73 @@
 ﻿using System;
 using System.IO;
+using System.Text;
+using System.Text.RegularExpressions;
 using System.Xml.Serialization;
+using EhsnPlugin.Mappers;
+using FieldDataPluginFramework;
+using FieldDataPluginFramework.Context;
+using FieldDataPluginFramework.Results;
 
 namespace EhsnPlugin
 {
     public class Parser
     {
+        private readonly IFieldDataResultsAppender _appender;
+        private readonly ILog _logger;
+ 
+        public Parser(IFieldDataResultsAppender appender, ILog logger)
+        {
+            _appender = appender;
+            _logger = logger;
+        }
+
         public EHSN LoadFromStream(Stream stream)
         {
             try
             {
-                var serializaer = new XmlSerializer(typeof(EHSN));
+                var cleanedUpXml = GetXmlWithEmptyElementsRemoved(stream);
 
-                var eHsn = serializaer.Deserialize(stream) as EHSN;
+                var serializaer = new XmlSerializer(typeof(EHSN));
+                var memoryStream = new MemoryStream((new UTF8Encoding()).GetBytes(cleanedUpXml));              
 
-                return eHsn;
+                return serializaer.Deserialize(memoryStream) as EHSN;
             }
             catch (Exception e)
             {
                 return null;
             }
         }
+
+        private string GetXmlWithEmptyElementsRemoved(Stream stream)
+        {
+            using (var streamReader = new StreamReader(stream, Encoding.UTF8,
+                detectEncodingFromByteOrderMarks: true, bufferSize: 4096, leaveOpen: true))
+            {
+                var originalXml = streamReader.ReadToEnd();
+                stream.Position = 0;
+
+                return Regex.Replace(originalXml, @"<[a-zA-Z][^<>]*\/>",string.Empty);
```

Comment:
That wasn't the regex I was expecting. I had thought `@"<[a-zA-Z0-9_]+ ?/>"` would have been enough.

Your regex looks like it might capture a bit more, including something like `<foo attribute="value"/>`. Is that what is needed?

Reply:
That's needed. For example
 ``` 		
	<ObservedVal row="1"/> 
 ```

My regex should include underscore *_* as the initial letter of an element name.